include/roboteam_utils/Line.h

```diff
@@ -14,176 +14,196 @@ class LineSegment;
+     * @return true True if this->slope() == line.slope()
+     * @return false False if this->slope() != line.slope()
+     */
+    [[nodiscard]] bool isParallel(const Line &line) const;
+    /**
+     * @brief Checks whether 2 lines are parralel
+     *
+     * @param line Other line to check against
+     * @return true True if this->slope() == line.slope()
+     * @return false False if this->slope() != line.slope()
+     */
+    [[nodiscard]] bool isParallel(const LineSegment &line) const;
+    /**
+     * @brief Checks whether line is a single point
+     *
+     * @return true True if start == end
+     * @return false False if start != end
+     */
+    [[nodiscard]] bool isPoint() const;
+    /**
+     * @brief Gets the distance from \ref point to the line
+     *
+     * @param point Point to get distance to
+     * @return double Distance to line
+     */
+    [[nodiscard]] double distanceToLine(const Vector2 &point) const;
+
+    /**
+     * @brief Checks whether a \ref point is on the line
+     *
+     * @param point Point to check `this` against
+     * @return true If \ref point is on `this`
+     * @return false If \ref point is not on `this`
+     */
+    [[nodiscard]] bool isOnLine(const Vector2 &point) const;
+
+    /**
+     * @brief Projects the point onto the line
+     *
+     * @param point Point to get the projection to
+     * @return Vector2 Vector projection from the line to the point
+     */
+    [[nodiscard]] Vector2 project(const Vector2 &point) const;
+
+    /**
+     * @brief Gets the intersection of the lines
+     * See https://en.wikipedia.org/wiki/Line%E2%80%93line_intersection for help
+     *
+     * @param line Line to get an intersection from
+     * @return std::shared_ptr<Vector2> Vector representation of this intersection
+     */
+    [[nodiscard]] std::optional<Vector2> intersects(const Line &line) const;
+
+    /**
+     * @brief Gets the intersection of the lines
+     * See https://en.wikipedia.org/wiki/Line%E2%80%93line_intersection for help
+     *
+     * @param line LineSegment to get an intersection from
+     * @return std::shared_ptr<Vector2> Vector representation of this intersection
+     */
+    [[nodiscard]] std::optional<Vector2> intersects(const LineSegment &line) const;
+
+    /**
+     * @brief Compute the line intersection of this line and the given as parameter line in positive direction.
+     * Which means in the direction from the start to the end of the given parameter line. The intersection position does not have to be in positive direction compared to this
+     * line, only to the given parameter line. Note if the line given as parameter is equal to this line then no intersection is returned.
+     *
+     * @param line The line of which the positive direction intersection is computed.
+     * @return No vector if the lines do not intersect in positive direction. Otherwise it returns a vector which represents the position of intersection.
+     */
```

Comment:
```suggestion
     Compute the intersection of this, and the line given as parameter. The parameter line is treated as half-line, so the a + tb, with b the direction vector, any point on the line where t<0 is not included in the calculation of the intersection. 
```
This description might not be correct for the way we implemented it but I think the general way of describing it is clearer. I still don't understand what this documentation is talking about, without having you explain it to me.